fix: use injected LoggerService in ConfigStorageService + fast follows twenty config

ehconitin opened the pull request:
Replaces direct Logger instantiation in ConfigStorageService with injected LoggerService to ensure consistent log handling across the application. This fixes the issue where ConfigStorageService error logs were appearing in integration test output while other services' logs were properly suppressed.

ehconitin commented on packages/twenty-server/src/engine/core-modules/twenty-config/storage/config-storage.service.ts:
```diff
@@ -21,13 +22,12 @@ import { ConfigStorageInterface } from './interfaces/config-storage.interface';
 
 @Injectable()
 export class ConfigStorageService implements ConfigStorageInterface {
-  private readonly logger = new Logger(ConfigStorageService.name);
```
I am either blind or an idiot :)

ehconitin commented:
<img width="889" alt="Screenshot 2025-05-17 at 13 13 31" src="https://github.com/user-attachments/assets/11028281-6e20-47c5-8eb8-37957cbadaec" />


I dont get why is this change causing integration tests to not show any logs at all, also its passing everything, but clickhouse test fails for me locally.

<img width="1025" alt="Screenshot 2025-05-17 at 13 15 56" src="https://github.com/user-attachments/assets/920967c1-b790-496d-8260-85985b6e134f" />

and this is not local to me! same with github ci

This is strange, digging in more

ehconitin commented:
ps: It might be because of the initialization order, TwentyConfigModule is loaded very early in the application bootstrap process (it's part of CoreEngineModule). It's one of the first modules to be initialized because other modules depend on it for their configuration.

I think its a classic `did the egg came first or the chicken` problem

FelixMalfait commented:
Gave feedback on Discord, I think we might be on a wrong path here and maybe it was about fixing the API so that it sends bad requests and don't log errors for expected API calls

ehconitin commented on packages/twenty-utils/dangerfile.ts:
```diff
@@ -18,19 +18,47 @@ if (packageChanged && !lockfileChanged) {
+      
+      // If we have added variables but fewer metadata decorators, likely missing metadata
+      if (addedVarLines.length > 0 && addedVarLines.length > addedMetadataLines.length) {
+        const message = 'It looks like you added new environment variables without the required @ConfigVariablesMetadata decorator';
```
wdyt? @FelixMalfait

FelixMalfait replied:
Not really convinced by the danger rule idea tbh.
It feels like it's something we could have checked when the module starts if we really want to.

FelixMalfait commented on packages/twenty-server/src/engine/core-modules/twenty-config/config-variables.ts:
```diff
@@ -637,7 +643,8 @@ export class ConfigVariables {
     type: ConfigVariableType.ARRAY,
-    options: ['log', 'error', 'warn', 'debug', 'verbose'],
+    options: ['log', 'error', 'warn', 'debug'],
+    isEnvOnly: true,
```
Are you sure we're not setting isEnvOnly on too many vars? It almost ruins the point of the admin panel to have so many

ehconitin replied:
Yes! I will be going thru them again

FelixMalfait commented on packages/twenty-server/src/engine/core-modules/twenty-config/config-variables.ts:
```diff
@@ -346,6 +348,7 @@ export class ConfigVariables {
     description: 'Type of storage to use (local or S3)',
     type: ConfigVariableType.ENUM,
     options: Object.values(StorageDriverType),
+    isEnvOnly: true,
```
This is the kind of variable that we built the admin panel for, would be a shame if they really don't work

FelixMalfait reviewed (commented):
Almost there!

charlesBochet commented:
Leaving it up to you, LGTM on the current code and agree with existing comments (also discussed with Felix offline)

charlesBochet commented:
@ehconitin, I'm going to close this PR for now to keep the history clean, please re-open once ready!

FelixMalfait commented on packages/twenty-server/src/engine/core-modules/twenty-config/dynamic-driver.base.ts:
```diff
@@ -0,0 +1,73 @@
+
+  protected abstract buildConfigKey(): string;
+  protected abstract createDriver(): TDriver;
+  protected cleanupDriver(_driver: TDriver): void {
```
I would advocate to remove this kind of thing since it's not used anywhere for now and the PR is fairly complex

charlesBochet commented on packages/twenty-server/src/engine/core-modules/twenty-config/dynamic-driver.base.ts:
```diff
@@ -0,0 +1,73 @@
+import { TwentyConfigService } from 'src/engine/core-modules/twenty-config/twenty-config.service';
+import { TypedReflect } from 'src/utils/typed-reflect';
+
+export abstract class DynamicDriverBase<TDriver> {
```
Naive question: what is a DynamicDriverBase?

charlesBochet commented on packages/twenty-server/src/engine/core-modules/email/email-sender.service.ts:
```diff
@@ -1,16 +1,88 @@
 @Injectable()
-export class EmailSenderService implements EmailDriver {
-  constructor(@Inject(EMAIL_DRIVER) private driver: EmailDriver) {}
+export class EmailSenderService
```
I'm a bit lost between EmailDriverInterface, DynamicDriverBase and EmailSenderService namings. Based on what I see in the file a more classic naming would be a Factory (a class returning a class depending on configuration)

Also, not sure we need this DynamicD

charlesBochet replied:
my comments are non blocking, I think this stays maintainable and it's better than what we have (non dynamic). Just adding my two cents

ehconitin commented:
Hey @charlesBochet! Thanks for the feedback - you're absolutely right about the naming and architecture concerns.

Why I originally avoided "Factory" naming - 
I was actually avoiding the word "factory" because I had just removed the module factories (the forRootAsync() functions), and in my head I was thinking "no more factories!" 😅 And the fact that I thought factories were meant only for modules for some reason

What I've implemented based on your feedback - 

I've refactored to address all your points:
1. Clear Factory Naming
- DynamicDriverBase -> DriverFactoryBase
- Created concrete factories: EmailDriverFactory and FileStorageDriverFactory

2. Services Use Composition (Factories Still Use Inheritance)
- Services changed: From extending DynamicDriverBase to injecting factory classes
- Factories still inherit: EmailDriverFactory extends DriverFactoryBase and FileStorageDriverFactory extends DriverFactoryBase
- Result: Services delegate to factories instead of having factory logic mixed into service logic

3. Separation of Concerns
- Factory responsibility: Create and cache drivers based on config
- Service responsibility: Delegate operations to current driver
